pkg/controller/deployment/deployment_controller.go

```diff
@@ -412,11 +414,23 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 
 	switch d.Spec.Strategy.Type {
 	case extensions.RecreateDeploymentStrategyType:
-		return dc.syncRecreateDeployment(d)
+		if err := dc.syncRecreateDeployment(d); err != nil {
+			return err
+		}
 	case extensions.RollingUpdateDeploymentStrategyType:
-		return dc.syncRollingUpdateDeployment(d)
+		if err := dc.syncRollingUpdateDeployment(d); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("unexpected deployment strategy type: %s", d.Spec.Strategy.Type)
+	}
+
+	// Cleanup old rcs
+	if d.Spec.RevisionHistoryLimit != nil {
```

Comment:
I think we agreed that this should be executed inside each strategy after scaling has finished.